lib/ansible/modules/monitoring/zabbix/zabbix_import.py

```diff
@@ -0,0 +1,260 @@
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "screens:"
+            - "  createMissing: true"
+            - "  updateExisting': true"
+            - "templateLinkage:"
+            - "  createMissing: true"
+            - "templates:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "templateScreens:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "triggers:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+            - "  deleteMissing: true"
+            - "valueMaps:"
+            - "  createMissing: true"
+            - "  updateExisting: true"
+
+
+extends_documentation_fragment:
+    - zabbix
+'''
+
+EXAMPLES = '''
+- name: Import zabbix hosts
+  local_action:
+    module: zabbix_import
+    server_url: http://monitor.example.com
+    login_user: username
+    login_password: password
+    import_file: /tmp/myfile.xml
+    import_format: xml
+    rules:
+      hosts:
+        createMissing: True
+    timeout: 10
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from zabbix_api import ZabbixAPI
+
+    HAS_ZABBIX_API = True
+except ImportError:
+    HAS_ZABBIX_API = False
+
+
+DEFAULT_RULES = ({
+    'applications': {
+        'createMissing': True,
+        'deleteMissing': True
+    },
+    'discoveryRules': {
+        'createMissing': True,
+        'updateExisting': True,
+        'deleteMissing': True
+    },
+    'graphs': {
+        'createMissing': True,
+        'updateExisting': True,
+        'deleteMissing': True
+    },
+    'groups': {
+        'createMissing': True
+    },
+    'hosts': {
+        'createMissing': True,
+        'updateExisting': True
+    },
+    'images': {
+        'createMissing': True,
+        'updateExisting': True
+    },
+    'items': {
+        'createMissing': True,
+        'updateExisting': True,
+        'deleteMissing': True
+    },
+    'maps': {
+        'createMissing': True,
+        'updateExisting': True
+    },
+    'screens': {
+        'createMissing': True,
+        'updateExisting': True
+    },
+    'templateLinkage': {
+        'createMissing': True
+    },
+    'templates': {
+        'createMissing': True,
+        'updateExisting': True
+    },
+    'templateScreens': {
+        'createMissing': True,
+        'updateExisting': True,
+        'deleteMissing': True
+    },
+    'triggers': {
+        'createMissing': True,
+        'updateExisting': True,
+        'deleteMissing': True
+    },
+    'valueMaps': {
+        'createMissing': True,
+        'updateExisting': True
+    }
+})
+
+
+class Configuration(object):
+    def __init__(self, module, zbx):
+        self._module = module
+        self._zapi = zbx
+
+    def import_template(self, filename, rules, import_format):
+        f = open(filename, 'r')
```

Comment:
I generally see a pattern among other modules to avoid opening files directly. They instead use lookup plugins like [file](https://docs.ansible.com/ansible/latest/plugins/lookup/file.html) (see 2nd example in [zabbix_template module](https://docs.ansible.com/ansible/latest/modules/zabbix_template_module.html#examples) or [authorized_key module](https://docs.ansible.com/ansible/latest/modules/authorized_key_module.html)).

First thing that comes to my mind is support for roles. Lets say I have files/import.xml and tasks/main.yml directories in them, then I would need to write task using your module like this:
```yaml
- zabbix_import:
    import_file: roles/role1/files/import.xml
    import_format: xml
```

and the goal is to refer to files/ directory implicitly:
```yaml
- zabbix_import:
    import_file: '{{ lookup("file", "import.xml") }}'
    import_format: xml
```

Other thing may be that lookup plugins are evaluated on Ansible control machine, so there is no need to copy XML with import data to machine where you are delegating your task to. See the [doc](https://docs.ansible.com/ansible/2.5/plugins/lookup.html) for more information.

I am sure that more experienced ansible maintainers than me would be able to bring another arguments for using lookup plugin here.